csharp/Worker/Microsoft.Spark.CSharp/Worker.cs

```diff
@@ -425,7 +425,9 @@ private object[] GetNext(int messageLength)
                     {
                         Debug.Assert(messageLength > 0);
                         byte[] buffer = SerDe.ReadBytes(inputStream, messageLength);
-                        result = unpickler.loads(buffer) as object[];
+                        var unpickledObjects = PythonSerDe.GetUnpickledObjects(buffer);
```

Comment:
I do not know the recommended usage for Unpickler and hence going with the current implementation. Pickling and unpickling is no longer a black box with the update to 1.5.2. So it is important to understand what is under the hood and make necessary perf tuning. That is one of the top items in my todo list. Hopefully I can provide a better response to your comment as I get started on that.